src/react-components/room/useMicrophone.js

```diff
@@ -69,8 +71,16 @@ export function useMicrophone(scene) {
     [mediaDevicesManager]
   );
 
+  useEffect(() => {
+    if (canVoiceChat) {
+      mediaDevicesManager.startMicShare({});
```

Comment:
Doesn't seem right to tie the actual behavior of mic toggling to a react hook. That would mean if no UI happens to be showing the uses `useMicrophone` this would not be toggled. Should just be handled outside of react land. Probably in MediaDevicesManager in the same spot we are toggling it off on permission changes?

Reply:
This is mostly because the useMicrophone hook is used in MicSetupModal and the audio popover as that's where we want to conditionally start the mic based on if the voice permission is enabled. I don't have a strong opinion about this but I've moved it back to where we had it before, in ui-root in case that makes more sense.

Reply:
Having MicSetupModal have an effect that initially enables the mic makes sense, since that's part of what that screen does. But having it here means this ends up being what is re-enabling the mic in response to permission changes which seems wrong. MediaDeviceManager is what handles turning disabling it when permissions change, so I would think in the same spot it should also handle re-enabling it.

Reply:
There are two different things. One is mic sharing and another one is mic enabling. Mic sharing just triggers the sharing process and ask for permission if needed. Enabling pauses/resumes the producer so we are effectively sending voice data.

In `ui-root` we are listening to voice permission updates to start sharing the mic during the different entry steps.
In the `MediaDevicesManager` we are listening to voice permission updates to pause the mic producer and we rely on the user to manually resume it as it might be tricky if the voice permission is re-enabled by a moderator and that forces the user's mic to open.

Reply:
Ah right, that makes sense. We should still probably move this effect out of `useMicrophone` though, since if anything happens to be using it when a permission change happens its going to trigger, since it has `canVoiceChat` as a dependency, and `useCan` listens for permission change events.